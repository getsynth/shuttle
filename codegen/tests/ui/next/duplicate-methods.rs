shuttle_codegen::app! {
    #[shuttle_codegen::endpoint(method = get, route = "/hello")]
    async fn hello() -> &'static str {
        "Hello, World!"
    }

    #[shuttle_codegen::endpoint(method = post, route = "/hello")]
    async fn goodbye() -> &'static str {
        "Goodbye, World!"
    }

    #[shuttle_codegen::endpoint(method = post, route = "/hello")]
    async fn goodbye() -> &'static str {
        "Goodbye, World!"
    }
}
