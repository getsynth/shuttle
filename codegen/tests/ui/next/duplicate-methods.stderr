error: only one method of each type is allowed per route

         = help: Remove one of the post methods on the "/hello" route.

  --> tests/ui/next/duplicate-methods.rs:12:42
   |
12 |     #[shuttle_codegen::endpoint(method = post, route = "/hello")]
   |                                          ^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/next/duplicate-methods.rs:16:2
   |
16 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/next/duplicate-methods.rs`
